src/directives/__tests__/input-restriction-uppercasealphanum.directive.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, DebugElement } from '@angular/core';
+import { InputRestrictionUppercaseAlphanumDirective } from '../input-restriction-uppercasealphanum.directive';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  template: `<input type="text" uppercaseAlphanumOnly>`,
+})
+class TestAlphaNumComponent {
+}
+
+describe('Directive: InputRestrictionUppercaseAlphanumDirective', () => {
+  let fixture: ComponentFixture<TestAlphaNumComponent>;
+  let inputElement: DebugElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TestAlphaNumComponent,
+        InputRestrictionUppercaseAlphanumDirective,
+      ],
+    });
+    fixture = TestBed.createComponent(TestAlphaNumComponent);
+  });
+  it('should remove spaces, non alpha numeric chars and upper case input', () => {
```

Comment:
test description made me think it would remove upper case - possibly change to 'should upper lowercase letters and remove spaces, and non alpha numeric chars' (or something like that)

Reply:
updated test description, thanks @rossfauldsbjss